akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazySourceSpec.scala

```diff
@@ -37,16 +37,14 @@ class LazySourceSpec extends StreamSpec with DefaultTimeout with ScalaFutures {
     }
 
     "never construct the source when there was no demand" in assertAllStagesStopped {
-      val probe = TestSubscriber.probe[Int]()
       val constructed = new AtomicBoolean(false)
       Source
         .lazySingle { () =>
           constructed.set(true)
           1
         }
-        .toMat(Sink.fromSubscriber(probe))(Keep.left)
```

Comment:
ok, maybe fromSubscriber is eagerly pulling? normally one would use `TestSink`, right?

Easier to use `Sink.cancelled` here, indeed.